features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+- Start Date: 2020-03-01
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+- Protocol Name: Out of Band
+- Version: 1.0
+- URI: `http://didcomm.org/out-of-band/VER/MsgType`
+
+## Summary
+
+The Out Of Band protocol is used when you wish to engage with another agent and you don't have a DIDComm connection to use for the interaction.
+
+## Motivation
+
+The use of the `invitation` in the [Connection](../../features/0160-connection-protocol/README.md) and [DID Exchange](../../features/0023-did-exchange/README.md) protocols has been relatively successful, but has some shortcomings, as follows.
+
+### Connection Reuse
+
+A common pattern we have seen in the early days of Aries agents is a user with a browser getting to a point where a connection is needed between the website's (enterprise) agent and the user's mobile agent. A QR invitation is displayed, scanned and a protocol is executed to establish a connection. Life is good!
+
+However, with the current invitation processes, when the same user returns to the same page, the same process is executed (QR code, scan, etc.) and a new connection is created between the two agents. There is no way for the user's agent to say "Hey, I've already got a connection with you. Let's use that one!"
+
+We need the ability to reuse a connection.
+
+### Connection Establishment Versioning
+
+In the existing Connections and DID Exchange `invitation` handling, the _inviter_ dictates what connection establishment protocol all _invitee_'s will use. A more sustainable approach is for the _inviter_ to offer the _invitee_ a list of support protocols and allow the _invitee_ to use one that it supports. This pattern enables the community deprecation of protocol versions.
+
+### Handling of all Out of Band Messages
+
+We currently have two sets of out of band messages that cannot be delivered via DIDComm because there is no channel. We'd like to align those message into a single "out of band" protocol so that their handling can be centralized, and a common QR code handling mechanism can be used.
+
+### URLs and QR Code Handling
+
+We'd like to have the specification of QR handling centralized into a single RFC (this one).
+
+## Tutorial
+
+### Key Concepts
+
+The out of band protocol is used when you don't know if you have a connection with another user. This could be because you are trying to establish a new connection with that agent, you have connections but don't know who the other party is, or if you want to have a connection-less interaction. Since there is no DIDComm connection to use for the messages of this protocol, the messages are plaintext and sent out of band, such as via a QR code. Since the delivery of out of band messages will often be via QR codes, this RFC also covers the use of QR codes.
```

Comment:
Can we give one other example besides QR codes, so it's clear that's not the only use case? How about email? ("such as via email or a QR code")

Reply:
I'll think about that. I'm not certain what the email would look like :-). I guess just 

Dear Daniel,

Click [here](http://example.com/ssi?c_i=eyAiQHR5cGUiOiAiaHR0cHM6Ly9kaWRjb21tLm9yZy9vdXRvZmJhbmQvMS4wL2ludml0YXRpb24iLCAiQGlkIjogIjY5MjEyYTNhLWQwNjgtNGY5ZC1hMmRkLTQ3NDFiY2E4OWFmMyIsICJsYWJlbCI6ICJBbGljZSIsICJwcm90b2NvbHMiOiBbImh0dHBzOi8vZGlkY29tbS5vcmcvZGlkZXhjaGFuZ2UvMS4wIiwgImh0dHBzOi8vZGlkY29tbS5vcmcvY29ubmVjdGlvbnMvMS4wIl0sICJzZXJ2aWNlIjogWyJkaWQ6c292OkxqZ3BTVDJyanNveFllZ1FEUm03RUwiXSB9) to connect.

Click here to get an agent...

```

Reply:
Added.  Also fixed the "centralized" reference. There was another that I changed as well.

Reply:
I don't see any example other than QR codes. Am I missing something? I searched for a fix...

Reply:
See example email message directly above [this heading](https://github.com/hyperledger/aries-rfcs/blob/95135828ce722f61cbc6dae3633f8b38f441c8aa/features/0434-outofband/README.md#url-shortening), below the QR code.

If you were looking for something else, please be a little more specific of what you want to see.

Reply:
Thank you for pointing me to it. I did genuinely look before I asked, but I guess I didn't look enough. Sorry to make you point out the obvious. The example is great.